spec/system/placements/schools/placements/school_user_edits_a_placement_with_no_mentors_spec.rb

```diff
@@ -0,0 +1,202 @@
+require "rails_helper"
+
+RSpec.describe "School user edits a placement with no mentors", service: :placements, type: :system do
+  scenario do
+    given_that_placements_exist
+    and_i_am_signed_in
+
+    when_i_am_on_the_placements_index_page
+    then_i_see_my_placement
+
+    when_i_click_on_my_placement
+    then_i_see_the_placement_details_page
+    and_i_see_the_add_a_mentor_link
+
+    when_i_click_on_add_a_mentor
+    then_i_see_the_mentors_index_page
+
+    when_i_click_on_add_mentor
+    then_i_see_the_find_teacher_page
+
+    when_i_fill_in_the_details
+    and_i_click_on_continue
+    then_i_see_the_confirm_details_page
+
+    when_i_click_on_confirm_and_add_mentor
+    then_i_see_the_mentors_index_page
+    and_i_see_a_success_message
+
+    when_i_click_on_the_placements_navigation_item
+    then_i_see_the_placements_index_page
+
+    when_i_click_on_my_placement
+    then_i_see_the_placement_details_page_with_assign_a_mentor_text
+  end
+
+  private
+
+  def given_that_placements_exist
+    @springfield_elementary_school = build(
+      :placements_school,
+      name: "Springfield Elementary",
+      address1: "Westgate Street",
+      address2: "Hackney",
+      postcode: "E8 3RL",
+      group: "Local authority maintained schools",
+      phase: "Primary",
+      gender: "Mixed",
+      minimum_age: 3,
+      maximum_age: 11,
+      religious_character: "Does not apply",
+      admissions_policy: "Not applicable",
+      urban_or_rural: "(England/Wales) Urban major conurbation",
+      percentage_free_school_meals: 15,
+      rating: "Outstanding",
+    )
+
+    @primary_english_subject = build(:subject, name: "Primary with english", subject_area: :primary)
+
+    @autumn_term = build(:placements_term, name: "Autumn term")
+    @spring_term = create(:placements_term, name: "Spring term")
+    @summer_term = create(:placements_term, name: "Summer term")
+
+    @current_academic_year_name = Placements::AcademicYear.current.name
+
+    @placement = create(
+      :placement,
+      school: @springfield_elementary_school,
+      subject: @primary_english_subject,
+      year_group: :year_1,
+      terms: [@autumn_term],
+    )
+
+    stub_valid_teaching_record_response
+  end
+
+  def and_i_am_signed_in
+    sign_in_placements_user(organisations: [@springfield_elementary_school])
+  end
+
+  def when_i_am_on_the_placements_index_page
```

Comment:
I mentioned this to Dan. I think it might be worth having a `current_path` check, just to ensure we're in the right place.

Reply:
I don't think that's necessarily valuable here, we know we're on the right page given the other checks and the user doesn't care about the URL, if we need to change our routes in the future I wouldn't expect a system test to break but I would want a request spec to break if that makes sense?

Reply:
Ah, yeah sorry! I thought this was the standard edit flow but it's not. I will remove 👍